lib/train/platforms/detect/helpers/os_windows.rb

```diff
@@ -2,7 +2,12 @@ module Train::Platforms::Detect::Helpers
   module Windows
     def detect_windows
       # try to detect windows, use cmd.exe to also support Microsoft OpenSSH
+      # setting pty to false to execute platform dection code as setting pty to true results in getting unwanted charters in response.
+      # resetting the same once output is gathered.
+
+      set_pty_false if @backend.class.to_s == "Train::Transports::SSH::Connection"
```

Comment:
That is some really brittle tight coupling.

Likely run_command needs to take some kind of `may_need_pty_false: true` flag (note: but some better name) which  `Train::Transports::SSH::Connection` implements and which the other connection types ignore.